ledger-core/instrumentation/insapp/server.go

```diff
@@ -15,26 +15,36 @@ import (
 	"github.com/insolar/component-manager"
 
 	"github.com/insolar/assured-ledger/ledger-core/configuration"
+	"github.com/insolar/assured-ledger/ledger-core/cryptography"
+	"github.com/insolar/assured-ledger/ledger-core/cryptography/keystore"
 	"github.com/insolar/assured-ledger/ledger-core/instrumentation/inslogger"
 	"github.com/insolar/assured-ledger/ledger-core/instrumentation/trace"
 	"github.com/insolar/assured-ledger/ledger-core/log/global"
+	"github.com/insolar/assured-ledger/ledger-core/network/nodeinfo"
 	"github.com/insolar/assured-ledger/ledger-core/vanilla/throw"
 	"github.com/insolar/assured-ledger/ledger-core/version"
 )
 
+type CertManagerFactory func(ctx context.Context, certPath string, comps PreComponents) nodeinfo.CertificateManager
+type KeyStoreFactory func(path string) (cryptography.KeyStore, error)
+
 type Server struct {
-	cfgPath string
-	appFn   AppFactoryFunc
-	multiFn MultiNodeConfigFunc
-	extra   []interface{}
+	cfgPath            string
+	appFn              AppFactoryFunc
+	multiFn            MultiNodeConfigFunc
+	extra              []interface{}
+	certManagerFactory CertManagerFactory
+	keyStoreFactory    KeyStoreFactory
```

Comment:
please make CertManagerFactory and KeyStoreFactory as a single functions of "ConfigurationProvider" that can return config, key store and cert manager

it will also be more compatible with PLAT-779 approach

Reply:
rewrote with 'ConfigurationProvider'

Reply:
then why do you keep'em as variables on the server?

Reply:
now only config provider is stored